chromoscores/snipping.py

```diff
@@ -0,0 +1,197 @@
+import numpy as np
+
+
+def peak_snipping(contact_map, window_size, peak_coordinate):
+    """
+    Peak snippet; snippet with a "window_size" around peak with "peak_coordinate"
+    on a contact map, "contact_map". peak_coordinate should be in the format
+     of (i,j)
+    -----------------
+    Function peak_snippet(contact_map, window_size, peak_coordinate):
+
+    begin function
+
+         raise an error if peak coordinate + window window_size is out of the map
+
+         snippet = contact_map[
+        (peak_coordinate[0] - window_size):(peak_coordinate[0] + window_size),
+        (peak_coordinate[1] - window_size):(peak_coordinate[1] + window_size)]
+
+     return snippet_matrix
+
+     end function
+    ----------------
+    """
+
+    if (peak_coordinate[1] + window_size) > len(contact_map) or (
+        peak_coordinate[0] - window_size
+    ) < 0:
+
+        raise ValueError(
+            "window window_size for peak coordinate exceeds window_size of the contact map"
+        )
+
+    snippet = contact_map[
+        (peak_coordinate[0] - window_size) : (peak_coordinate[0] + window_size),
+        (peak_coordinate[1] - window_size) : (peak_coordinate[1] + window_size),
+    ]
+
+    return snippet
+
+
+def tad_snipping(contact_map, stall_list, index):
+    """
+    Tad snippet
+    To extract Tad snippets adjacent to a boundary element with index of
+    "index" from a list of "stall_list":
```

Comment:
if you want to include the first TAD does the boundary list start at 0? if not this should probably be mentioned

should an error arise if max(boundary_list) > len(contact_map)? should an error arise if min(boundary_list) < 0?

Reply:
It can start from 0. I think the condition of

 "    if index + 1 > len(boundary_list):
        raise ValueError("index + 1 should be in the range of boundary list")"

 should resolve the issue. boundary list is supposed to be within the length of the contact map.